sopel/modules/admin.py

```diff
@@ -56,7 +56,11 @@ def join(bot, trigger):
         return
     elif not key:
         bot.join(channel)
+        if channel not in bot.config.core.channels:
+            bot.config.core.channels = bot.config.core.channels + [channel]
+            bot.config.save()
     else:
+        # TODO: a way to store a channel's key in configuration
```

Comment:
Does this help?

https://github.com/sopel-irc/sopel/blob/af308621198f4e825063efbe472f8ec42d53e5d4/sopel/bot.py#L259-L266

Reply:
Basically, you should be able to just store (in `bot.config.core.channels`) `' '.join(channel, key)` and have it work after restart unless the channel key changes.

Reply:
Ooooooooh... that is super weird, but yes, it helps!

Reply:
It's a little weird, but that's also how IRC handles channel keys. `JOIN #channel key` is how a client joins a single channel with a key. (Sopel doesn't deal with more than one channel at a time, so the proper IRC syntax for multiple channels—`JOIN #chan1,#chan2 key1,key2`—isn't relevant… yet.)